tests/unit/dataset/test_dataset_manager.py

```
@@ -843,3 +844,113 @@ def test_chat_does_not_require_inline_media(self):
+        await manager.initialize()
+        return manager
+
+    async def test_random_sampling_raises_service_error(self) -> None:
+        """Explicit random sampling is rejected in accuracy mode."""
+        user_config = _make_accuracy_user_config(
+            strategy=DatasetSamplingStrategy.RANDOM
+        )
+        manager = await self._make_manager(user_config)
+
+        with pytest.raises(
+            ServiceError, match="random.*not supported|not supported.*random"
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)